src/features/author/authorSlice.tsx

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+
+export interface AuthorState {
+  author: User | null;
```

Comment:
Change 'User | null' to 'User | undefined' to make it more consistent with other parts of the codebase where optional properties are used.